ghost/stripe/lib/StripeAPI.js

```diff
@@ -222,6 +222,30 @@ module.exports = class StripeAPI {
         return customer;
     }
 
+    /**
+     * Finds a Stripe Customer ID based on the provided email address. Returns null if no customer is found.
+     * @param {string} email
+     * @see https://stripe.com/docs/api/customers/search
+     *
+     * @returns {Promise<string|null>} Stripe Customer ID, if found
+     */
+    async getCustomerIdByEmail(email) {
+        try {
+            const result = await this._stripe.customers.search({
+                query: `email:"${email}"`,
+                limit: 1
+            });
```

Comment:
I'm wondering if we shouldn't search more than 1 customer, and return the customer with the newest subscriptions? To avoid reusing an old customer

Reply:
Yes that makes sense, thanks! Updated in [300364f](https://github.com/TryGhost/Ghost/pull/17326/commits/300364f83606933356a91df8025e72b87ee5da41)